<?xml version="1.0"?>
<robot name="Puzzlebot Jetson-Lidar Ed.">

    <link name="base_link"></link>

    <!-- Chassis Joint -->
    <joint name="chassis_joint" type="fixed">
        <parent link="base_link"/>
        <child link="chassis"/>
        <origin xyz="0.0 0.0 0.05"/>
    </joint>

    <link name="chassis">
        <inertial>
            <origin xyz="0 0 0" rpy="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="7.58333333333e-06" ixy="0.0" ixz="0.0" iyy="7.58333333333e-06" iyz="0.0" izz="1.25e-05"/>
        </inertial>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://rviz_simulation/models/MCR2_1000_13_Chassis.stl"/>
            </geometry>
        </collision>
        <visual>
            <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0 0"/>
            <geometry>
                <mesh filename="package://rviz_simulation/models/MCR2_1000_13_Chassis.stl"/>
            </geometry>
        </visual>
    </link>

    
    <!-- Left Wheel Joint -->
    <joint name="leftWheel" type="continuous">
        <origin xyz="0.0525 0.103 0.0" rpy="0 0.0 0.0"/>
        <parent link="chassis"/>
        <child link="wl_link"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- Left Wheel Link -->
    <link name="wl_link">
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy = "1.57079632679 0 0"/>
            <geometry>
                <mesh filename="package://rviz_simulation/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
        </visual>
    </link>

    <!-- Right Wheel Joint -->
    <joint name="rightWheel" type="continuous">
        <parent link="chassis"/>
        <child link="wr_link"/>
        <origin xyz="0.0525 -0.103 0.0"/>
        <axis xyz="0 1 0"/>
    </joint>

    <!-- Right Wheel Link -->
    <link name="wr_link">
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy= "-1.57079632679 0 0"/>
            <geometry>
                <mesh filename="package://rviz_simulation/models/MCR2_1000_1_1_Wheel_Coupler_2.stl"/>
            </geometry>
        </visual>
    </link>

    
    <!-- Camera Joint -->
    <joint name="camera_joint" type="fixed">
        <parent link="chassis"/>
        <child link="camera_link"/>
        <origin xyz="0.1 0.0 0.04" rpy="-1.57079632679 0 -1.57079632679"/>
    </joint>

    <!-- Camera Link -->
    <link name="camera_link">
        <visual>
            <origin xyz="0 0.0 0.0" rpy="0 0 3.14159"/>
            <geometry>
                <mesh filename="package://rviz_simulation/models/camera.stl"/>
            </geometry>
        </visual>
    </link>

    <joint name="aruco_joint" type="fixed">
        <origin xyz="0.0 0.0 50.0" rpy="0.0 0.0 0.0"/>
        <parent link="base_link"/>
        <child link="aruco_box"/>
        <axis xyz="0 1 0"/>
    </joint>

    <link name="aruco_box">
        <visual>
            <origin xyz="-0.025 0.025 -0.025" rpy = "0 0 0"/>
            <geometry>
                <mesh filename="package://rviz_simulation/models/cube.stl" scale="0.0025 0.0025 0.0025"/>
            </geometry>
        </visual>
    </link>

</robot>
